Layer handling simplified, and streamlined

algernon opened the pull request:
This simplifies the layer handling by only allowing a single layer to be active at any one time. As such, the behaviour after this patch should be as described, and as discussed in #15, as far as I understand.

Also moved the layer handling to a separate file, let me know if that is not desired.

Fixes #15, if all goes well.

algernon committed:
Simplify the layer handling

There can only be one layer active at any one time, simplify the code
accordingly.

Signed-off-by: Gergely Nagy <algernon@madhouse-project.org>

algernon committed:
Move layer handling to a separate file

Signed-off-by: Gergely Nagy <algernon@madhouse-project.org>

mondalaci commented on right/src/layer.c:
```diff
@@ -0,0 +1,11 @@
+
+uint8_t ActiveLayer = LAYER_ID_BASE;
+
+void Layer_MoveTo(uint8_t layer) {
```
I would rather leave out these helper functions. They don't seem to add to the readability but make our internal API more complex.

algernon replied:
A'ight. Update coming up in a bit then.

algernon committed:
Move layer handling back to keyboard_layout.c

And while there, also drop the helper functions, and just do the activation in
the handler function. This simplifies the code a bit, as there is little point
in lifting out an assignment into a function at this time.

Signed-off-by: Gergely Nagy <algernon@madhouse-project.org>

algernon committed:
Move ActiveLayer back to keyboard_layout.c too

That is the only place we use it from, no need to expose it outside of that.

Signed-off-by: Gergely Nagy <algernon@madhouse-project.org>

algernon commented:
Updated, should be much simpler now.

mondalaci commented:
Looks great, thanks!
